library/std/src/process.rs

```diff
@@ -1741,6 +1741,28 @@ impl ExitCode {
     /// behavior][exit#platform-specific-behavior]). `ExitCode` exists because of this; only
     /// `ExitCode`s that are valid on all platforms can be created, so those problems don't exist
     /// with this method.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use std::process::ExitCode;
+    /// # use std::fmt;
+    /// # enum UhOhError { GenericProblem, Specific, WithCode { exit_code: ExitCode, _x: () } }
+    /// # impl fmt::Display for UhOhError {
+    /// #     fn fmt(&self, _: &mut fmt::Formatter) -> fmt::Result { unimplemented!() }
+    /// # }
+    /// // there's no way to gracefully recover from an UhOhError, so we just
+    /// // print a message and exit
+    /// fn handle_unrecoverable_error(err: UhOhError) -> ! {
+    ///     eprintln!("UH OH! {err}");
+    ///     let code = match err {
+    ///         UhOhError::GenericError => ExitCode::FAILURE,
+    ///         UhOhError::Specific => ExitCode::from(3),
+    ///         UhOhError::WithCode { exit_code, .. } => exit_code,
+    ///     };
```

Comment:
looks good, though as soon as we land generic member access I'd love to update this to use that, so instead it becomes:

`err.context::<ExitCode>().unwrap_or(ExitCode::FAILURE).exit_process();`

Reply:
Should I add a `/// # // TODO:` or something? Or oh I guess probably just `// TODO` lol